client/src/main/java/org/spine3/client/QueryFactory.java

```diff
@@ -0,0 +1,192 @@
+
+    private static final String ENTITY_IDS_EMPTY_MSG = "Entity ID set must not be empty";
+
+    private final ActorContext actorContext;
+
+    QueryFactory(ActorRequestFactory actorRequestFactory) {
+        checkNotNull(actorRequestFactory);
+        this.actorContext = actorRequestFactory.actorContext();
+    }
+
+    private static QueryId newQueryId() {
+        final String formattedId = format(QUERY_ID_FORMAT, Identifiers.newUuid());
+        return QueryId.newBuilder()
+                      .setValue(formattedId)
+                      .build();
+    }
+
+    /**
+     * Creates a new instance of {@link QueryBuilder} for the further {@linkplain Query}
+     * construction.
+     *
+     * @param targetType the {@linkplain Query query} target type
+     * @return new instance of {@link QueryBuilder}
+     */
+    public QueryBuilder select(Class<? extends Message> targetType) {
+        checkNotNull(targetType);
+        final QueryBuilder queryBuilder = new QueryBuilder(targetType, this);
+        return queryBuilder;
+    }
+
+    /**
+     * Creates a {@link Query} to read certain entity states by IDs with the {@link FieldMask}
+     * applied to each of the results.
+     *
+     * <p>Allows to specify a set of identifiers to be used during the {@code Query} processing.
+     * The processing results will contain only the entities, which IDs are present among
+     * the {@code ids}.
+     *
+     * <p>Allows to set property paths for a {@link FieldMask}, applied to each of the query
+     * results. This processing is performed according to the
+     * <a href="https://goo.gl/tW5wIU">FieldMask specs</a>.
+     *
+     * <p>In case the {@code paths} array contains entries inapplicable to the resulting entity
+     * (e.g. a {@code path} references a missing field),
+     * such invalid paths are silently ignored.
+     *
+     * @param entityClass the class of a target entity
+     * @param ids         the entity IDs of interest
+     * @param maskPaths   the property paths for the {@code FieldMask} applied
+     *                    to each of results
+     * @return an instance of {@code Query} formed according to the passed parameters
+     */
+    public Query byIdsWithMask(Class<? extends Message> entityClass,
+                               Set<? extends Message> ids,
+                               String... maskPaths) {
+        checkNotNull(ids);
+        checkArgument(!ids.isEmpty(), ENTITY_IDS_EMPTY_MSG);
+
+        final FieldMask fieldMask = FieldMask.newBuilder()
+                                             .addAllPaths(Arrays.asList(maskPaths))
+                                             .build();
+        final Query result = composeQuery(entityClass, ids, null, fieldMask);
+        return result;
+    }
+
+    /**
+     * Creates a {@link Query} to read certain entity states by IDs.
+     *
+     * <p>Allows to specify a set of identifiers to be used during the {@code Query} processing.
```

Comment:
Done for this particular case. When talking about the results of the query we're currently building, let's prefer Future Simple if possible.